file_catalog/server.py

```diff
@@ -227,16 +229,52 @@ def get(self):
     @coroutine
     def post(self):
         metadata = json_decode(self.request.body)
-        ret = yield self.db.get_file({'file_name':metadata['file_name']})
+
+        # check metadata for mandatory fields
+        if not all(x in metadata for x in ['locations', 'uid', 'checksum']):
```

Comment:
I think this is better written as
```python
if not set(('uid','checksum','locations')).issubset(metadata):
```

But if you don't like the set syntax, then just convert the list to a tuple in the current line.